ext/date/date_core.c

```diff
@@ -7432,6 +7436,98 @@ d_lite_jisx0301(VALUE self)
     return strftimev(fmt, self, set_tmx);
 }
 
+static VALUE
+deconstruct_keys(VALUE self, VALUE keys, int is_datetime) {
+    VALUE h;
+    long i;
+
+    get_d1(self);
+
+    if (NIL_P(keys)) {
+        h = rb_hash_new();
```

Comment:
You could do `VALUE h = rb_hash_new();` and eliminate the two lines for setting `h`, since they are now the same.